### DISCUSSION TITLE
Add charging features

### DISCUSSION THREAD

**Discussion by darthcalamus** [bimmerconnected/bimmer_connected#435] [Feature request] [answered]
Hi,
the BEVs support additional charging features. This would be extremely helpful to introduce "smart charging" to dumb chargers.

charge mode (immediate, timer based) => OK
SoC goal - 20%-100%
AC Power Limit - 6A-32A

Can you please add this to the backlog?

And thanks for the awesome work!

**Comment by fredriklj** [comment]
Especially SoC goal would be awesome to have. Please let me know if there is anything I can do to assist (recent owner of an i4).

**Comment by gerard33** [comment]
Step 1 is done via #516. Next step is to add charging mode, charging preference, ac charging limit and charging target (SoC goal) to HA.

**Comment by gerard33** [comment]
The first [PR](https://github.com/home-assistant/core/pull/88759#pullrequestreview-1314588710) for HA is waiting for review. After that is merged I will make a PR to add the entities described above.

In the meanwhile you can use the custom [integration ](https://github.com/bimmerconnected/ha_custom_component)which already contains all of the new goodies 😄

**Comment by rikroe** [comment, marked as answer]
Implemented since HA 2023.6 ✔️
